src/mongo/mongoConnectionStrings.ts

```diff
@@ -3,19 +3,55 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-export function getDatabaseNameFromConnectionString(connectionString: string): string | undefined {
-    //  Connection strings follow the following format (https://docs.mongodb.com/manual/reference/connection-string/):
-    // mongodb://[username:password@]host1[:port1][,host2[:port2],...[,hostN[:portN]]][/[database][?options]]
+function parseConnectionString(connectionString: string): string[] | undefined {
+    // Connection strings follow the following format (https://docs.mongodb.com/manual/reference/connection-string/):
+    //   mongodb[+srv]://[username:password@]host1[:port1][,host2[:port2],...[,hostN[:portN]]][/[database][?options]]
     // Some example connection strings:
-    //   mongodb://dbuser:dbpassword@dbname.mlab.com:14118/dbname
-    //   mongodb://db1.example.net:27017,db2.example.net:2500/?replicaSet=test
-    //   mongodb://router1.example.com:27017,router2.example2.com:27017,router3.example3.com:27017/
+    //   mongodb://dbuser:dbpassword@dbname.mlab.com:14118
+    //   mongodb+srv://db1.example.net:27017,db2.example.net:2500/?replicaSet=test
+    //   mongodb://router1.example.com:27017,router2.example2.com:27017,router3.example3.com:27017/database?ssh=true
+    // Regex splits into five parts:
+    //   Full match
+    //   mongodb[+srv]
+    //   [username:password@]host1[:port1][,host2[:port2],...[,hostN[:portN]]]
+    //   [database]
+    //   [?options]
+
+    try {
+        let matches = connectionString.match('(^mongodb(?:[+]srv)?):\/\/([^\/]*)(?:\/([^/?]*)(.*))?$');
+        if (matches && matches.length === 5) {
+            return matches.map(str => {
+                if (str === "") {
+                    str = undefined;
+                }
+                return str;
+            });
+        }
+    } catch (error) {
+        // Shouldn't happen, but ignore if does
+    }
+
+    return undefined;
+}
+
+export function getDatabaseNameFromConnectionString(connectionString: string): string | undefined {
+    try {
+        let [, , , database] = parseConnectionString(connectionString);
+        return database;
+    } catch (error) {
+        // Shouldn't happen, but ignore if does
+    }
+
+    return undefined;
+}
 
+export function addDatabaseToConnectionString(connectionString: string, databaseName: string): string | undefined {
     try {
-        let matches = connectionString.match(/[a-zA-Z]+:\/\/[^/]+(\/([^/?]+))/);
-        if (matches) {
-            let [, , database] = matches;
-            return database;
+        let [, , , parsedDatabaseName] = parseConnectionString(connectionString);
+        if (!parsedDatabaseName) {
+            return connectionString.replace(new RegExp('(^mongodb(?:[+]srv)?:\/\/[^\/]*)(?:\/((?:[^/?]*).*)?|$)'), `$1\/${databaseName}$2`);
```

Comment:
Why change from `[a-zA-Z]+` to `(^mongodb(?:[+]srv)?`?

Why change from `[^\/]+` to `[^\/]*`?

Why do you have the second capturing group in the replace?

We want to keep this regexp as close to the existing as possible and I don't think those changes are necessary.

Reply:
1. `[a-zA-Z]+` is not the exact pattern for MongoDB connection string, it should be ```mongodb``` or ```mongodb+srv```.
2. `[^\/]*` parses strings like this: `mongodb:///?replicaSet=test` as `["mongodb://", ""(undefined), ""(undefined), "?replicaSet=test"]`, but `[^\/]+` doesn't parse. Not sure which behavior is better 
3. Options are the part of mongo connection string, so I thought that we also should keep it for database connection strings.

Actually, I have the same pattern in `parseConnectionString`

Reply:
1. Can you link to documentation for 1? You're making the pattern more strict and I want to make sure we're not breaking users with connections strings we haven't thought of.
1. I don't think `mongodb:///?replicaSet=test` is a valid connection string
1. Try a test without the capturing group. The 'replace' method only replaces the matching part and should keep the rest of the string (the options) intact

Reply:
From a quick rereading of the docs (in the comments), I don't think the mongodb+srv scheme allows a database name. Do you disagree?

Reply:
Seems like you're right, couldn't find any docs that say about mongo+srv connection string with a database too. Is it possible to attach or maybe even create a database account with connection string of `mongodb+srv` format inside cosmos?

Reply:
@PrashanthCorp You're the Cosmos expert now. Do you know how to test a connection string with mongodb+srv?

Reply:
Can we keep the original `[a-zA-Z]+` and ignore (or file an issue for) the 'mongodb+srv' scenario? I feel like we're going down a rabbit hole that I'm not convinced anyone is using (and I don't want to test). In general I think it would be better for @Omelianenko to keep the regex as similar to the original as possible and focus on the App Service scenario rather than Cosmos DB edge cases.

Reply:
It's for sure one of the two possible variants: `mongodb `or `mongodb+srv` ([link](https://docs.mongodb.com/manual/reference/connection-string/)), so it's not hard to make `mongodb(?:[+]srv)` instead of `[a-zA-Z]+`, but it's not a big deal at all, so just going to leave `[a-zA-Z]+`

Reply:
I'm not worried about it. In an ideal world @Omelianenko should file an issue to test the "mongodb+srv" scenario before release, but at this point I just care about getting the core scenarios merged with minimal risk of regressions in Cosmos DB.